### PR TITLE
Added workflow for semantic versioning

### PR DESCRIPTION
**Pull request by Jawayria**
The objective of proposed workflow is to automate the process of versioning. This workflow will trigger whenever a commit os pushed on master branch. It will perform these steps:
1. Based on the commit type, suggest the next version
2. Compare the suggested version with latest tag to ensure the version is going to be bumped
3. Update the version in repo
4. Tag the commit with new version number
5. Append new changes in the CHANGELOG.rst
6. Create GitHub release and generate a PR

### CODE REVIEW

File: .github/workflows/changelogs.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    - name: Determine Version
+      id: gitversion
+      run: echo "::set-output name=version::$(semantic-release print-version)"
+
+    - name: Get Latest Tag
+      id: latesttag
+      run: echo "::set-output name=tag::$(git describe --tags --abbrev=0)"
+
+    - name: Update Version
+      if: ${{ !contains(steps.gitversion.outputs.version,steps.latesttag.outputs.tag) }}
```
Comment (nedbat) [inline .github/workflows/changelogs.yml]: This conditional appears a lot.  What is it checking?

**Comment by nedbat** [comment]
Can you write up a description of what this is doing and how it works in the pull request description?

File: .github/workflows/changelogs.yml (hunk 1 above)
Comment (Jawayria) [reply .github/workflows/changelogs.yml]: It checks whether Gitversion suggests a version upgrade or it points to the already created tag. If a version upgrade is suggested, only then further steps execute.

**Comment by jmbowman** [comment]
@nedbat Not sure if you got a notification, but it looks like the description here has been updated as requested.

File: .github/workflows/changelogs.yml (hunk 1 above)
Comment (nedbat) [reply .github/workflows/changelogs.yml]: Maybe we can simplify this by using two jobs, and having the second one have one "if" that tests the outputs from the first job?  https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs

File: changelogs.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import re
```
Comment (nedbat) [inline changelogs.py:1]: Is the plan for this to be used by more repos?

File: changelogs.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+# Save new changelogs in temp.md
+changelogs_cmd = "semantic-release changelog"
+process = subprocess.Popen(changelogs_cmd.split(), stdout=subprocess.PIPE)
+output, error = process.communicate()
+output = "## [" + sys.argv[1] + "] - " + str(date.today()) + "\n"+output.decode("utf-8")
+
+file = open('temp.md', 'w')
+file.writelines(output)
+file.close()
```
Comment (nedbat) [inline changelogs.py]:
```suggestion
with open('temp.md', 'w') as f:
    f.write(output)
```

File: changelogs.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+# Convert the changelogs to rst
+pandoc_cmd = "pandoc temp.md -f markdown_strict -t rst -o temp.rst"
+process = subprocess.Popen(pandoc_cmd.split(), stdout=subprocess.PIPE)
+output, error = process.communicate()
```
Comment (nedbat) [inline changelogs.py]: Perhaps you should check the error status, or use `check_output` if you are only going to use the output.

File: changelogs.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+file = open('temp.rst', 'r')
+changelogs = file.read().split('\n')
```
Comment (nedbat) [inline changelogs.py]:
```suggestion
with open('temp.rst') as f:
    changelogs = f.readlines()
```

File: changelogs.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+for line_num in range(len(changelogs)):
+    obj = re.search(r'``[a-zA-Z0-9]{7}``',  changelogs[line_num])
+    if obj:
+        changelogs[line_num] = re.sub(r'``[a-zA-Z0-9]{7}``', obj.group().split('`')[2], changelogs[line_num])
+changelogs = "\n".join(changelogs)
+
+
+# append the changelogs in CHANGELOG.rst
+for line in fileinput.FileInput("CHANGELOG.rst", inplace=1):
+    if ".. <New logs>" in line:
```
Comment (nedbat) [inline changelogs.py]: It would be safer to use `line.startswith(".. <New logs>")`

File: changelogs.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        line = line.replace(line, line+"\n"+changelogs)
+        print(line)
+    else:
+        print(line.split('\n')[0])
```
Comment (nedbat) [inline changelogs.py]: Is this to get rid of the trailing newline?  `line.rstrip('\n')` will be clearer.

File: changelogs.py (hunk 2 above)
Comment (Jawayria) [reply changelogs.py:1]: YES this script will be used by the 'changelogs' GitHub workflow and will be added in other repos as well

File: changelogs.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+from datetime import date
+import fileinput
+
+# Save new changelogs in temp.md
+changelogs_cmd = "semantic-release changelog"
+output = subprocess.check_output(changelogs_cmd.split())
+output = "## [" + sys.argv[1] + "] - " + str(date.today()) + "\n"+output.decode("utf-8")
+
+with open('temp.md', 'w') as file:
+    file.writelines(output)
```
Comment (nedbat) [inline changelogs.py]: `writelines` works here, but is confusing: writelines is documented to take a list of strings as an argument, but you are giving it a single string.  You should use `.write(output)` here.

**Comment by nedbat** [comment]
@Jawayria let me know when you are ready for another review.

**Comment by Jawayria** [comment]
@nedbat I've updated the PR. Ready for another review

**Review by nedbat — APPROVED** [review]

**Comment by Jawayria** [comment]
Actually I am incrementing the version number just for testing purposes

**Pull request merged**